alf/examples/sac_carla_conf.py

```diff
@@ -0,0 +1,206 @@
+# limitations under the License.
+
+from functools import partial
+import torch
+
+import alf
+from alf.algorithms.td_loss import TDLoss
+from alf.algorithms.data_transformer import ImageScaleTransformer, ObservationNormalizer, RewardNormalizer
+from alf.environments import suite_carla
+from alf.environments.carla_controller import VehicleController
+
+import carla_conf
+import sac_conf
+
+alf.config('ImageScaleTransformer', min=0.0, fields=['observation.camera'])
+alf.config('ObservationNormalizer', clipping=5.0)
+
+# training config
+alf.config(
+    'TrainerConfig',
+    initial_collect_steps=3000,
+    unroll_length=10,
+    mini_batch_size=64,
+    num_updates_per_train_iter=1,
+    num_iterations=1000000,
+    num_checkpoints=20,
+    evaluate=False,
+    debug_summaries=True,
+    summarize_grads_and_vars=True,
+    summary_interval=100,
+    replay_buffer_length=70000,
+    summarize_action_distributions=True,
+)
+
+tasac = False
+
+if tasac:
+    alf.config('RewardNormalizer', clip_value=1.0)
+    alf.config(
+        'TrainerConfig',
+        mini_batch_length=8,
+        data_transformer_ctor=[
+            ImageScaleTransformer, ObservationNormalizer, RewardNormalizer
+        ])
+else:
+    alf.config(
+        'TrainerConfig',
+        mini_batch_length=4,
+        data_transformer_ctor=[ImageScaleTransformer, ObservationNormalizer])
+
+alf.config(
+    'suite_carla.Player',
+    # Not yet able to successfully train with sparse reward.
+    sparse_reward=False,
+    # Cannot train well with allow_negative_distance_reward=False yet.
+    allow_negative_distance_reward=True,
+    # Currently, even a small penalty such as one make the training much worse
+    max_collision_penalty=20.,
+    max_red_light_penalty=20.,
+
+    # uncomment the following line to use PID controller
+    # controller_ctor=VehicleController,
+    with_gnss_sensor=False,
+)
+
+alf.config(
+    'CarlaEnvironment',
+    vehicle_filter='vehicle.audi.tt',
+    num_other_vehicles=20,
+    num_walkers=20,
+    # 1000 second day length means 4.5 days in replay buffer of 90000 length
+    day_length=1000,
+    max_weather_length=500,
+)
+
+encoding_dim = 256
+fc_layers_params = (256, )
+activation = torch.relu_
+use_batch_normalization = False
+learning_rate = 1e-4
+
+proj_net = partial(
+    alf.networks.StableNormalProjectionNetwork,
+    state_dependent_std=True,
+    squash_mean=False,
+    scale_distribution=True,
+    min_std=1e-3,
+    max_std=10)
+
+actor_network_cls = partial(
+    alf.networks.ActorDistributionNetwork,
+    fc_layer_params=fc_layers_params,
+    activation=activation,
+    use_fc_bn=use_batch_normalization,
+    continuous_projection_net_ctor=proj_net)
+
+env = alf.get_env()
+
+critic_network_cls = partial(
+    alf.networks.CriticNetwork,
+    joint_fc_layer_params=fc_layers_params,
+    activation=activation,
+    use_fc_bn=use_batch_normalization,
+    output_tensor_spec=env.reward_spec())
```

Comment:
maybe have a common.get_reward_spec() function, and hide alf.get_env() inside it?